pkg/kubelet/kubelet_test.go

```diff
@@ -2926,6 +2993,166 @@ func TestUpdateExistingNodeStatus(t *testing.T) {
+		Type:               api.NodeOutOfDisk,
+		Status:             api.ConditionFalse,
+		Reason:             "KubeletHasSufficientDisk",
+		Message:            fmt.Sprintf("kubelet has sufficient disk space available"),
+		LastHeartbeatTime:  unversioned.Time{}, // placeholder
+		LastTransitionTime: unversioned.Time{}, // placeholder
+	}
+
+	testCases := []struct {
+		rootFsAvail   uint64
+		dockerFsAvail uint64
+		expected      api.NodeCondition
+	}{
+		{
+			// NodeOutOfDisk==false
+			rootFsAvail:   200,
+			dockerFsAvail: 200,
+			expected:      ood,
+		},
+		{
+			// NodeOutOfDisk==true
+			rootFsAvail:   50,
+			dockerFsAvail: 200,
+			expected:      ood,
+		},
+		{
+			// NodeOutOfDisk==false
+			rootFsAvail:   200,
+			dockerFsAvail: 200,
+			expected:      ood,
+		},
+		{
+			// NodeOutOfDisk==true
+			rootFsAvail:   200,
+			dockerFsAvail: 50,
+			expected:      ood,
+		},
+		{
+			// NodeOutOfDisk==false
+			rootFsAvail:   200,
+			dockerFsAvail: 200,
+			expected:      noOod,
+		},
+	}
+
+	kubelet.updateRuntimeUp()
+	for tcIdx, tc := range testCases {
+		// Step by a second
+		clock.Step(1 * time.Second)
+
+		// Make kubelet report that it has sufficient disk space
+		if err := updateDiskSpacePolicy(kubelet, mockCadvisor, 500, 500, tc.rootFsAvail, tc.dockerFsAvail, 100, 100); err != nil {
+			t.Fatalf("can't update disk space manager: %v", err)
+		}
+
+		if err := kubelet.updateNodeStatus(); err != nil {
+			t.Errorf("unexpected error: %v", err)
+		}
+		actions := kubeClient.Actions()
+		if len(actions) != 2 {
+			t.Errorf("%d. unexpected actions: %v", tcIdx, actions)
+		}
+		updateAction, ok := actions[1].(testclient.UpdateAction)
+		if !ok {
+			t.Errorf("%d. unexpected action type.  expected UpdateAction, got %#v", tcIdx, actions[1])
+		}
+		updatedNode, ok := updateAction.GetObject().(*api.Node)
+		if !ok {
+			t.Errorf("%d. unexpected object type", tcIdx)
+		}
+		kubeClient.ClearActions()
+
+		var oodCondition api.NodeCondition
+		for i, cond := range updatedNode.Status.Conditions {
+			// Expect LastProbeTime to be updated to Now, while LastTransitionTime to be the same.
+			if clock.Since(cond.LastHeartbeatTime.Time) > 0 {
+				t.Errorf("%d. Condition %v LastProbeTime:\nexpected \n%s\n, got \n%s", tcIdx, cond.Type, clock.Now(), cond.LastHeartbeatTime.Time)
+			}
+			// In the last case, there should be a status transition for NodeOutOfDisk
+			if tcIdx == len(testCases)-1 && cond.Type == api.NodeOutOfDisk {
+				if clock.Since(cond.LastTransitionTime.Time) > 0 {
+					t.Errorf("%d. Condition %v LastTransitionTime:\nexpected \n%#v\n, got \n%#v", tcIdx, cond.Type, clock.Now(), cond.LastTransitionTime.Time)
+				}
+			} else {
+				if clock.Since(cond.LastTransitionTime.Time) == 0 {
+					t.Errorf("%d. Condition %v LastTransitionTime:\nexpected \n%#v\n, got \n%#v", tcIdx, cond.Type, clock.Now(), cond.LastTransitionTime.Time)
+				}
+			}
+			updatedNode.Status.Conditions[i].LastHeartbeatTime = unversioned.Time{}
```

Comment:
A few lines below, we do a deep equals between the objects. It is hard to compare time values.  So I followed this pattern that other tests use in this file.

Now that we can precisely control the time set by the tests, we can switch to deep equals based comparisons even for time values.